packages/sdk/src/conversation/middlewares/cardActionMiddleware.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  ActivityTypes,
+  CardFactory,
+  InvokeResponse,
+  MessageFactory,
+  Middleware,
+  StatusCodes,
+  TurnContext,
+} from "botbuilder";
+import { AdaptiveCardResponse, TeamsFxAdaptiveCardActionHandler } from "../interface";
+import { InvokeResponseFactory } from "../invokeResponseFactory";
+
+/**
+ * @internal
+ */
+export class CardActionMiddleware implements Middleware {
+  public readonly actionHandlers: TeamsFxAdaptiveCardActionHandler[] = [];
+  private readonly defaultMessage: string = "Your response was sent to the app";
+
+  constructor(handlers?: TeamsFxAdaptiveCardActionHandler[]) {
+    if (handlers && handlers.length > 0) {
+      this.actionHandlers.push(...handlers);
+    }
+  }
+
+  async onTurn(context: TurnContext, next: () => Promise<void>): Promise<void> {
+    if (context.activity.name === "adaptiveCard/action") {
+      const action = context.activity.value.action;
+      const actionVerb = action.verb;
+
+      for (const handler of this.actionHandlers) {
+        if (handler.triggerVerb === actionVerb) {
+          let response: InvokeResponse;
+          try {
+            response = await handler.handleActionInvoked(context, action.data);
+          } catch (error: any) {
+            const errorResponse = InvokeResponseFactory.errorResponse(
+              StatusCodes.INTERNAL_SERVER_ERROR,
+              error.message
+            );
+            await this.sendInvokeResponse(context, errorResponse);
+            throw error;
+          }
+
+          const responseType = response.body?.type;
+          switch (responseType) {
+            case "application/vnd.microsoft.activity.message":
+              await this.sendInvokeResponse(context, response);
+              break;
+            case "application/vnd.microsoft.card.adaptive":
+              const card = response.body?.value;
+              if (!card) {
+                await this.sendInvokeResponse(
+                  context,
+                  InvokeResponseFactory.textMessage(this.defaultMessage)
```

Comment:
Should an error response be returned here when the error 'Adaptive card content cannot be found in the response body' occurs? Otherwise, the end user might think the action is executed successfully however it's not...

Reply:
Sounds good~

Reply:
The invoke response here should be an error response rather than a response with default message. Will fix it.